src/pages/experiments/[experimentId]/data-exploration/components/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -21,18 +22,19 @@ const ComparisonType = Object.freeze({ between: 'between', within: 'within' });
 
 const DiffExprCompute = (props) => {
   const {
-    experimentId, onCompute, cellSets, diffExprType,
+    experimentId, onCompute,
   } = props;
 
   const dispatch = useDispatch();
 
   const properties = useSelector((state) => state.cellSets.properties);
   const hierarchy = useSelector((state) => state.cellSets.hierarchy);
   const [isFormValid, setIsFormValid] = useState(false);
-  const [selectedGroups, setSelectedGroups] = useState(cellSets);
-  const [type, setType] = useState(diffExprType || ComparisonType.between);
+  const comparisonGroup = useSelector((state) => state.differentialExpression.comparison.group);
+  const selectedComparison = useSelector((state) => state.differentialExpression.comparison.type);
 
-  // `between sampoles/groups` makes no sense if there is no metadata. First,
+
+  // `between samples/groups` makes no sense if there is no metadata. First,
   // assume there is and correct this if it's necessary.
```

Comment:
reading from the code, it seems like we first assume there ISN'T metadata. If that is indeed the case, can you fix this comment to be accurate?

Reply:
I don't understand this one either, I was just fixing a typo :)) But I'll try to dig what that means

Reply:
it is about the default value for `hasMetadata` boolean which in here is set to `false`:

```
const [hasMetadata, setHasMetadata] = useState(false);
```

Judging by the code, by default we assume that there is no metadata, but the comment above says the opposite, so it should be corrected.

Reply:
changed in commit 11a74c2